tests/state_consistency.rs

```diff
@@ -0,0 +1,275 @@
+}
+
+async fn send_message_with_insufficient_gas(
+    client: &GearApi,
+    listener: &mut EventListener,
+    destination: [u8; 32],
+    payload: impl Encode + Debug,
+) -> Result<String> {
+    Ok(
+        send_message_with_custom_limit::<()>(client, listener, destination, payload, |gas| {
+            gas - gas / 100
+        })
+        .await?
+        .expect_err("Received a reply instead of an error message"),
+    )
+}
+
+#[tokio::test]
+#[ignore]
+async fn state_consistency() -> Result<()> {
+    let client = GearApi::dev().await?;
+    let mut listener = client.subscribe().await?;
+
+    let storage_code_hash = upload_code(&client, "target/ft_storage.wasm").await?;
+    let ft_logic_code_hash = upload_code(&client, "target/ft_logic.wasm").await?;
+
+    let ft_actor_id = upload_program(
+        &client,
+        &mut listener,
+        "target/ft_main.wasm",
+        InitFToken {
+            storage_code_hash,
+            ft_logic_code_hash,
+        },
+    )
+    .await?;
+
+    let (goc_actor_id, reply) = upload_program_and_wait_reply::<Result<(), GOCError>>(
+        &client,
+        &mut listener,
+        "target/wasm32-unknown-unknown/debug/game_of_chance.opt.wasm",
+        GOCInit {
+            admin: ALICE.into(),
+        },
+    )
+    .await?;
+    assert_eq!(reply, Ok(()));
+
+    let amount = 12345;
+
+    assert!(
+        FTokenEvent::Ok
+            == send_message(
+                &client,
+                &mut listener,
+                ft_actor_id,
+                FTokenAction::Message {
+                    transaction_id: 0,
+                    payload: Action::Mint {
+                        recipient: ALICE.into(),
+                        amount,
+                    }
+                    .encode(),
+                },
+            )
+            .await?
+    );
+    assert!(
+        FTokenEvent::Ok
+            == send_message(
+                &client,
+                &mut listener,
+                ft_actor_id,
+                FTokenAction::Message {
+                    transaction_id: 1,
+                    payload: Action::Approve {
+                        approved_account: goc_actor_id.into(),
+                        amount,
+                    }
+                    .encode(),
+                },
+            )
+            .await?
+    );
+
+    println!(
+        "{:?}",
+        send_message::<Result<GOCEvent, GOCError>>(
+            &client,
+            &mut listener,
+            goc_actor_id,
+            GOCAction::Start {
+                duration: 15000,
+                participation_cost: 10000,
+                ft_actor_id: Some(ft_actor_id.into())
+            }
+        )
+        .await?
+    );
+
+    let mut payload = GOCAction::Enter;
+
+    println!(
+        "{}",
+        send_message_with_insufficient_gas(&client, &mut listener, goc_actor_id, payload).await?
+    );
+    // TODO: add an equality check for reply after after introducing an
```

Comment:
This function has been added to `gclient` in the master branch: https://docs.gear.rs/gclient/struct.GearApi.html#method.last_block_timestamp